clusterdock/models.py

```diff
@@ -44,10 +45,19 @@ class Cluster:
 
     Args:
         *nodes: One or more :py:obj:`clusterdock.models.Node` instances.
+        name (:obj:`str`, optional): Cluster name to use. Default: :py:func:`generate_cluster_name`
     """
 
-    def __init__(self, *nodes):
-        self.name = generate_cluster_name()
+    def __init__(self, *nodes, name=None):
+        if name:
+            clusters = [container.cluster_name for container in get_containers(clusterdock=True)]
```

Comment:
Might as well make this is a set from the start to avoid having to do the cast in the subsequent line:
```
clusters = {container.cluster_name for container in get_containers(clusterdock=True)}
```